### PR TITLE
ui update for signup, login and chat screen

### PR DESCRIPTION
**Pull request by akinolu52**
fix for [PR #2](https://github.com/akinolu52/turbo-rotary/issues/2)

Screenshot 2021-08-06 at 21 55 39
Screenshot 2021-08-06 at 21 55 46
Screenshot 2021-08-06 at 21 56 08

### CODE REVIEW

File: client/src/Login.js [hunk 1]
```diff
@@ -8,12 +8,68 @@ import {
...
 import { login } from "./store/utils/thunkCreators";
+import Layout from './components/Layout';
+
+const useStyles = makeStyles(() => ({
+  right: {
+    // height: '100%',
+    position: 'relative',
+    display: 'flex',
+    // flexDirection: "column",
+    padding: '2rem 6rem',
```
Comment (FilipeGorodscy) [inline client/src/Login.js]: You can remove commented-out code from your PRs

File: client/src/Login.js [hunk 2]
```diff
@@ -23,55 +79,67 @@ const Login = (props) => {
...
+                    endAdornment: (
+                      <InputAdornment position='end' className={classes.labelFocusStyle}>
+                        <span>Forgot Password?</span>
+                      </InputAdornment>
+                    ),
+                  }}
+                />
+              </FormControl>
+            </Grid>
+            <div style={{ width: '90%', textAlign: 'center' }}>
```
Comment (FilipeGorodscy) [inline client/src/Login.js]: You should not mix JSX with Material UI and you should avoid inline styles.
All `<div>` occurrences should be changed for `<Box>` or `<Grid>` (depending on what properties you might want)

File: client/src/Login.js [hunk 3]
```diff
@@ -23,55 +79,67 @@ const Login = (props) => {
...
+                  name="password"
+                  required
+                  InputLabelProps={{
+                    className: classes.floatingLabelFocusStyle,
+                  }}
+                  // endAdornment={<InputAdornment position="end">Forgot</InputAdornment>}
+                  InputProps={{
+                    endAdornment: (
+                      <InputAdornment position='end' className={classes.labelFocusStyle}>
+                        <span>Forgot Password?</span>
```
Comment (FilipeGorodscy) [inline client/src/Login.js]: `<span>` should be changed to `<Typography>`

File: client/src/components/Sidebar/CurrentUser.js [hunk 4]
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
...
+  const handleClick = () => {
+    setOpen((prev) => !prev);
+  };
+
+  const handleClickAway = () => {
+    setOpen(false);
+  };
+
+  const handleLogout = async () => {
+    console.log('yes')
```
Comment (FilipeGorodscy) [inline client/src/components/Sidebar/CurrentUser.js]: You can remove console logs from your PRs

File: client/src/components/Sidebar/CurrentUser.js [hunk 5]
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
...
+    marginTop: '2.2rem'
+  },
 }));
 
 const CurrentUser = (props) => {
   const classes = useStyles();
+  const [open, setOpen] = useState(false);
+  let { logoutUser, user } = props;
+
+  user = user || {};
```
Comment (FilipeGorodscy) [inline client/src/components/Sidebar/CurrentUser.js]:
```suggestion
  const { logoutUser, user={} } = props;
```

File: client/src/components/Sidebar/CurrentUser.js [hunk 6]
```diff
@@ -29,20 +31,67 @@ const useStyles = makeStyles(() => ({
...
+            <MoreHorizIcon classes={{ root: classes.ellipsis }} onClick={handleClick} />
+            {open ? (
+              <div className={classes.dropdown}>
+                <Button className={classes.button1}
+                  onClick={handleLogout}
+                >
+                  Logout
+                </Button>
+              </div>
+            ) : null}
```
Comment (FilipeGorodscy) [inline client/src/components/Sidebar/CurrentUser.js]:
```suggestion
            {open && (
              <div className={classes.dropdown}>
                <Button className={classes.button1}
                  onClick={handleLogout}
                >
                  Logout
                </Button>
              </div>
            )}
```

**Review by FilipeGorodscy — COMMENTED** [review]
Good job overall. Some notes:
- It looks like some of your font colours (especially the phrase on the top right corner) are a little off compared to the mock design.
- There are a lot of repeated styles/components between Login/Signup pages.
- It looks like the linear gradient in the background image for Login/Signup pages is different from the mock design.

I'll attach a screenshot of the mobile view of the application so the other reviewers can reference it. Please let us know on Slack when this PR is ready for a second review!
Screen Shot 2021-08-06 at 5 24 58 PM

File: client/src/Login.js (hunk 1 above)
Comment (akinolu52) [reply client/src/Login.js]: okay

File: client/src/components/Sidebar/CurrentUser.js (hunk 5 above)
Comment (akinolu52) [reply client/src/components/Sidebar/CurrentUser.js]: okay, thanks

**Comment by akinolu52** [comment]
PR updated

Screenshot 2021-08-07 at 16 16 36

File: client/src/Login.js [hunk 7]
```diff
@@ -1,19 +1,40 @@
...
+    width: '90%',
+    textAlign: 'center',
+    marginTop: '2.2rem',
+    [theme.breakpoints.down('md')]: {
+      width: '100%',
+    },
+  },
+  button: {
+    color: '#FFF',
+    backgroundColor: "#3A8DFF",
```
Comment (cassandrale179) [inline client/src/Login.js]: Nit: we can use theme here for color as well

File: client/src/Signup.js [hunk 8]
```diff
@@ -1,118 +1,92 @@
...
+    width: '90%',
+    textAlign: 'center',
+    marginTop: '2.2rem',
+    [theme.breakpoints.down('md')]: {
+      width: '100%',
+    },
+  },
+  button: {
+    color: '#FFF',
+    backgroundColor: "#3A8DFF",
```
Comment (cassandrale179) [inline client/src/Signup.js]: Same here

File: client/src/components/Layout/AuthLayout.js [hunk 9]
```diff
@@ -0,0 +1,123 @@
...
+import bubble from '../../images/bubble.svg';
+
+const useStyles = makeStyles((theme) => ({
+    container: {
+        minHeight: '100vh',
+        [theme.breakpoints.down('sm')]: {
+            flexDirection: 'column'
+        },
+    },
+    left: {
```
Comment (cassandrale179) [inline client/src/components/Layout/AuthLayout.js]: Nit: left is not a descriptive name, let's change this to .SideBar or something similar.

File: client/src/components/Layout/AuthLayout.js [hunk 10]
```diff
@@ -0,0 +1,123 @@
...
+        marginTop: '20px',
+        [theme.breakpoints.down('sm')]: {
+            fontSize: '1.7rem'
+        },
+        [theme.breakpoints.down('xs')]: {
+            fontSize: '1.2rem',
+            marginBottom: '1rem'
+        },
+    },
+    right: {
```
Comment (cassandrale179) [inline client/src/components/Layout/AuthLayout.js]: Nit, same here, let's rename this class to be something more descriptive other than 'right'

File: client/src/components/Layout/AuthLayout.js [hunk 11]
```diff
@@ -0,0 +1,123 @@
...
+            padding: '1.5rem 3rem',
+            flex: 3,
+            flexDirection: 'column-reverse',
+            justifyContent: 'flex-end',
+        },
+        [theme.breakpoints.down('xs')]: {
+            padding: '1.2rem 1.5rem'
+        },
+    },
+    button1: {
```
Comment (cassandrale179) [inline client/src/components/Layout/AuthLayout.js]: Same here, let's rename this from button1 and button2 to what the button represent, such as 'FormButton' or 'HeaderButton'

File: client/src/components/Sidebar/CurrentUser.js [hunk 12]
```diff
@@ -29,20 +31,63 @@ const useStyles = makeStyles(() => ({
...
 
   return (
     <Box className={classes.root}>
       <BadgeAvatar photoUrl={user.photoUrl} online={true} />
       <Box className={classes.subContainer}>
         <Typography className={classes.username}>{user.username}</Typography>
-        <MoreHorizIcon classes={{ root: classes.ellipsis }} />
+
+        <ClickAwayListener onClickAway={handleClickAway}>
+          <div className={classes.dropdownRoot}>
```
Comment (cassandrale179) [inline client/src/components/Sidebar/CurrentUser.js]: To keep it consistent with the code base, let's not mix regular JSX expression with Material UI and used Grid or Box when appropriate.

**Review by cassandrale179 — APPROVED** [review]

File: client/src/Login.js (hunk 7 above)
Comment (akinolu52) [reply client/src/Login.js]: okay

File: client/src/components/Layout/AuthLayout.js (hunk 11 above)
Comment (akinolu52) [reply client/src/components/Layout/AuthLayout.js]: thanks

File: client/src/Signup.js (hunk 8 above)
Comment (akinolu52) [reply client/src/Signup.js]: thanks, done

**Pull request merged**